client/pages/product-details.jsx

```diff
@@ -0,0 +1,161 @@
+import React from 'react';
+import Container from 'react-bootstrap/Container';
+import parseRoute from '../lib/parse-route';
+import Row from 'react-bootstrap/Row';
+import ListGroup from 'react-bootstrap/ListGroup';
+import Accordion from 'react-bootstrap/Accordion';
+
+export default class ProductDetails extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      product: {},
+      route: parseRoute(window.location.hash),
+      loading: false
+    };
+    this.sizes = this.sizes.bind(this);
+  }
+
+  componentDidMount() {
+    const { params } = this.state.route;
+    const paramObject = Object.fromEntries(params);
+    const productId = paramObject.product;
+    fetch(`api/products/${productId}`)
+      .then(res => res.json())
+      .then(product => this.setState({ product, loading: true }))
+      .catch(err => console.error(err));
+    if (this.state.product === {}) {
+      this.setState({ loading: false });
+    }
+  }
+
+  sizes() {
+    const sizes = [];
+    for (let i = 0; i < this.state.product.length; i++) {
+      sizes.push(this.state.product[i].size);
+    }
+    const listItems = sizes.map(size => {
+      return <ListGroup.Item className="fs-5 border border-secondary d-inline-block w-auto mx-1 rounded my-3" key={size}>
+        <div>{size}</div>
+      </ListGroup.Item>;
+    });
+    return (
+      <ListGroup className="d-flex flex-row flex-wrap">
+        {listItems}
+      </ListGroup>
+    );
+  }
+
+  render() {
+    if (this.state.loading === true) {
+      const product = this.state.product[0];
```

Comment:
If we flip this condition, we can avoid nesting the rest of the code in the `if` block.

```js
if (this.state.loading) return null; // not ready, show nothing
```